src/Synthesiser.cpp

```diff
@@ -365,8 +365,7 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
         void visitClear(const RamClear& clear, std::ostream& out) override {
             PRINT_BEGIN_COMMENT(out);
 
-            out << "if (!isHintsProfilingEnabled()"
-                << (clear.getRelation().isTemp() ? ") " : "&& performIO) ");
+            out << "if (performIO) ";
```

Comment:
I think you also need the check to see the relation is a temp relation. Relations are purged if they are temp, or if performIO is true
```suggestion
            if (!clear.getRelation().isTemp()) {
                out << "if (performIO) ";
            }
```